astropy/cosmology/tests/test_connect.py

```diff
@@ -0,0 +1,184 @@
+
+cosmo_instances = cosmology.parameters.available
+save_formats = ["json"]
+
+
+###############################################################################
+# Setup
+
+def from_mapping(map, **kwargs):
+    params = copy.deepcopy(map)  # so can pop
+
+    # get cosmology. if string, parse to class
+    # 1st from 'kwargs'. Allows for override of the cosmology, if on file.
+    # 2nd from params. This MUST have the cosmology if 'kwargs' did not.
+    if "cosmology" in kwargs:
+        cosmology = kwargs.pop("cosmology")
+    else:
+        cosmology = params.pop("cosmology")
+    if isinstance(cosmology, str):
+        cosmology = _COSMOLOGY_CLASSES[cosmology]
+
+    # select arguments from mapping that are in the cosmo's signature.
+    ba = cosmology._init_signature.bind_partial()  # blank set of args
+    ba.apply_defaults()  # fill in the defaults
+    for k in cosmology._init_signature.parameters.keys():  # iter thru sig
+        if k in params:  # transfer argument, if in params
+            ba.arguments[k] = params.pop(k)
+
+    return cosmology(*ba.args, **ba.kwargs)
+
+
+def to_mapping(cosmology, *args, **kwargs):
+    m = {}
+    m["cosmology"] = cosmology.__class__
+    m.update({k: v for k, v in cosmology._init_arguments.items()
+              if k not in ("meta",)})
+    m["meta"] = copy.deepcopy(cosmology.meta)
+
+    return m
+
+
+def read_json(filename, key=None, **kwargs):
+    with open(filename, "r") as file:
+        data = file.read()
+    mapping = json.loads(data)  # parse json mappable to dict
+    # deserialize Quantity
+    for k, v in mapping.items():
+        if isinstance(v, dict) and "value" in v and "unit" in v:
+            mapping[k] = u.Quantity(v["value"], v["unit"])
+    return from_mapping(mapping, **kwargs)
+
+
+def write_json(cosmology, file, *, overwrite=False, **kwargs):
+    data = to_mapping(cosmology)  # start by turning into dict
+    data["cosmology"] = data["cosmology"].__name__  # change class field to str
+    # serialize Quantity
+    for k, v in data.items():
+        if isinstance(v, u.Quantity):
+            data[k] = {"value": v.value.tolist(),
+                       "unit": str(v.unit)}
+
+    # check that file exists and whether to overwrite.
+    if os.path.exists(file) and not overwrite:
+        raise IOError(f"{file} exists. Set 'overwrite' to write over.")
+    with open(file, "w") as write_file:
+        json.dump(data, write_file)
+
+
+def json_identify(origin, filepath, fileobj, *args, **kwargs):
+    return filepath is not None and filepath.endswith(".json")
+
+
+def setup_module(module):
+    """Setup module for tests."""
+    io_registry.register_reader("json", Cosmology, read_json)
+    io_registry.register_writer("json", Cosmology, write_json)
+    io_registry.register_identifier("json", Cosmology, json_identify)
+
+
+def teardown_module(module):
+    """clean up module after tests."""
+    with pytest.warns(FutureWarning):  # idk
```

Comment:
What does 'idk' stand for?

Reply:
"I don't know". It was an unexpected and seemingly unrelated warning. I think the warning occurs when the registry is completely cleared.

Reply:
Ah, that makes sense. Anyway, that clause will then soon disappear!